servers/account-data-deleter/src/dataService/exportStateService.ts

```diff
@@ -176,42 +216,55 @@ export class ExportStateService {
     payload: ExportPartComplete | ExportRequested,
   ): Promise<ExportRequestRecord | undefined> {
     const now = new Date();
-    if (payload['detail-type'] === PocketEventType.EXPORT_REQUESTED) {
+    try {
+      if (payload['detail-type'] === PocketEventType.EXPORT_REQUESTED) {
+        serverLogger.info({
+          message: 'ExportStateService - updateStatus - Creating new export request record',
+          body: payload.detail.requestId
+        });
+        const input = {
+          // Create new export request record
+          TableName: config.listExport.dynamoTable,
+          Key: {
+            requestId: payload.detail.requestId,
+          },
+          UpdateExpression: `SET expiresAt = :ea, createdAt = :ca`,
+          ExpressionAttributeValues: {
+            ':ea': Math.floor(now.getTime() / 1000) + 60 * 60 * 24 * 17, // 17 days (longer than parts retention),
+            ':ca': now.toISOString(),
+          },
+          ReturnValues: 'ALL_NEW' as const,
+        };
+        const result = await this.dynamo.send(new UpdateCommand(input));
+        return result.Attributes as ExportRequestRecord;
+      }
+      // Update the status of the export request to reflect
+      // completed compontents
+      // Remove illegal character
+      const service = payload.detail.service.replace('-', '');
       const input = {
-        // Create new export request record
         TableName: config.listExport.dynamoTable,
         Key: {
           requestId: payload.detail.requestId,
         },
-        UpdateExpression: `SET expiresAt = :ea, createdAt = :ca`,
+        ExpressionAttributeNames: { '#ss': service },
+        UpdateExpression: `SET #ss = :ss, ${service}CompletedAt = :sca`,
         ExpressionAttributeValues: {
-          ':ea': Math.floor(now.getTime() / 1000) + 60 * 60 * 24 * 17, // 17 days (longer than parts retention),
-          ':ca': now.toISOString(),
+          ':ss': true,
+          ':sca': payload.detail.timestamp,
         },
         ReturnValues: 'ALL_NEW' as const,
       };
       const result = await this.dynamo.send(new UpdateCommand(input));
       return result.Attributes as ExportRequestRecord;
+    } catch(error) {
```

Comment:
i think it would be better to create two `try/catch` blocks here - one for the creation of a new export request record, and one for the updating of the status of that record. it would help us get a better targeted error message.